chainercv/datasets/sbd/sbd_instance_segmentation_dataset.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import os
+
+import chainer
+import scipy
+
+from chainercv.datasets.sbd import sbd_utils
+from chainercv.utils import read_image
+
+
+class SBDInstanceSegmentationDataset(chainer.dataset.DatasetMixin):
+
+    """Instance segmentation dataset for Semantic Boundaries Dataset `SBD`_.
+
+    The class name of the label :math:`l` is :math:`l` th element of
+    :obj:`chainercv.datasets.voc_semantic_segmentation_label_names`.
+
+    .. _`SBD`: http://home.bharathh.info/pubs/codes/SBD/download.html
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/sbd`.
+        split ({'train', 'val', 'trainval'}): Select a split of the dataset.
+
+    """
+
+    def __init__(self, data_dir='auto', split='train'):
+        if split not in ['train', 'trainval', 'val']:
+            raise ValueError(
+                'please pick split from \'train\', \'trainval\', \'val\'')
+
+        if data_dir == 'auto':
+            data_dir = sbd_utils.get_sbd()
+
+        id_list_file = os.path.join(
+            data_dir, '{}_voc2012.txt'.format(split))
+        self.ids = [id_.strip() for id_ in open(id_list_file)]
+
+        self.data_dir = data_dir
+
+    def __len__(self):
+        return len(self.ids)
+
+    def get_example(self, i):
+        """Returns the i-th example.
+
+        Returns a color image, bounding boxes, masks and labels. The color
+        image is in CHW format.
+
+        Args:
+            i (int): The index of the example.
+
+        Returns:
+            tuple of color image, bounding boxes, masks and labels whose
+            shapes are (3, H, W), (N, 4), (N, H, W), (N, ) respectively.
+            H and W are height and width of the images, and N is the number
+            of objects in the image. The dtype of the color image and
+            the bounding boxes are :obj:`numpy.float32`, and that of the
+            masks and the labels are :obj:`numpy.int32`.
+
+        """
+        data_id = self.ids[i]
+        img_file = os.path.join(
+            self.data_dir, 'img', data_id + '.jpg')
+        img = read_image(img_file, color=True)
+        seg_img, ins_img = self._load_seg_ins(data_id)
+        bbox, mask, label = self._prepare_data(seg_img, ins_img)
+        return img, bbox, mask, label
+
+    def _load_seg_ins(self, data_id):
+        seg_file = os.path.join(
+            self.data_dir, 'cls', data_id + '.mat')
+        ins_file = os.path.join(
+            self.data_dir, 'inst', data_id + '.mat')
+        seg_mat = scipy.io.loadmat(seg_file)
```

Comment:
Can you import `scipy` like this?
https://github.com/chainer/chainercv/blob/master/chainercv/visualizations/vis_image.py#L19

Reply:
OK, i will add warnings too

Reply:
OK, i will add warnings too